paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
-
-        if self.backend == "amber":
-            # Generate `frcmod` and dummy atom files.
-            raise NotImplementedError
+    @classmethod
+    def prepare_host_structure(
+        cls, coordinate_path: str, host_atom_indices: Optional[List[int]] = None
+    ) -> pmd.Structure:
+        """Attempts to align the cavity of the host along the z-axis ready
+        for the APR calculation.
+
+        Parameters
+        ----------
+        coordinate_path
+            The path to the coordinate file which contains the host molecule.
+        host_atom_indices
+            The indices of the host atoms in the coordinate file if the file. This may
+            be used if the coordinate file contains more than a single molecule.
+        """
+        import parmed.geometry
+
+        # noinspection PyTypeChecker
+        structure = pmd.load_file(coordinate_path, structure=True)
 
-        self.directory.mkdir(parents=True, exist_ok=True)
-        installed_benchmarks = get_benchmarks()
-        if additional_benchmarks is not None:
-            installed_benchmarks.update(additional_benchmarks)
-
-        host_yaml, guest_yaml = self.parse_yaml(installed_benchmarks, guest_orientation)
-
-        self.benchmark_path = host_yaml.parent
-        self.host_yaml = read_yaml(host_yaml)
-        if guest:
-            self.guest_yaml = read_yaml(guest_yaml["yaml"])
-
-        if build:
-            # Here, we build desolvated windows and pass the files to the OpenFF Evaluator.
-            # These files are stored in `self.desolvated_window_paths`.
-            self.build_desolvated_windows(guest_orientation)
-            if generate_gaff_files:
-                generate_gaff(mol2_file=self.benchmark_path.joinpath(self.host_yaml["structure"]),
-                              residue_name=self.host_yaml["resname"],
-                              output_name=self.host,
-                              directory_path=self.directory,
-                              gaff=gaff_version)
-                if guest:
-                    generate_gaff(mol2_file=self.benchmark_path.joinpath(
-                        self.guest).joinpath(self.guest_yaml["structure"]),
-                                  output_name=self.guest,
-                                  residue_name=self.guest_yaml["name"],
-                                  directory_path=self.directory,
-                                  gaff=gaff_version)
-        if not build:
-            self.populate_window_list(input_pdb=os.path.join(self.directory, f"{self.host}-{self.guest}.pdb" if self.guest is not None
-            else f"{self.host}.pdb"))
-
-    def parse_yaml(self, installed_benchmarks, guest_orientation):
+        # Extract the host from the full structure.
+        if not host_atom_indices:
+            host_atom_indices = range(len(structure.atoms))
+
+        host_structure = structure[
+            "@" + ",".join(map(lambda x: str(x + 1), host_atom_indices))
```

Comment:
Is this 1-based numbering going to work for both AMBER and OpenMM? 

SInce this is a new function, we probably don't have any tests covering this.

Reply:
These are indices passed to ParmEd which uses 1-based indices I think so here we have the +1. The input indices should be zero based - I clarified this in the docstring. But yes this will work with both OMM + Amber as ParmEd is used as the intermediate.

Reply:
👍